src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.cs

```diff
@@ -317,7 +341,7 @@ public void Dispose()
         /// </remarks>
         public async ValueTask DisposeAsync()
         {
-            if (_output == null)
+            if (_output == null && _stream == null)
```

Comment:
Nit: Is there a reason both of these do:
```C#
if (_output == null && _stream == null)
```
whereas `CheckIfDisposed` does:
```C#
if (_stream == null)
{
    if (_output == null)
    {
        ...
    }
}
```
?

Reply:
Nope, just making sure reviewers are paying attention ;)

I thought about this discrepancy, but didn't worry about it too much since Dispose should only ever be called once "per workload" whereas Flush/etc. would be more common (particularly during grow).

Will fix.